### ISSUE TITLE
New-AzAKS - New-Azaks : The value of parameter orchestratorProfile.OrchestratorVersion is invalid.

### ISSUE THREAD

**Issue by lukeb1961** [Azure/azure-powershell#8526]
first attempt to use New-AKS. As per HELP examples. Using latest AZ module
PS > (get-module -name AZ).Version
Major  Minor  Build  Revision
-----  -----  -----  --------
1      3      0      -1      
PS> (get-module -name AZ.aks).Version
Major  Minor  Build  Revision
-----  -----  -----  --------
1      0      1      -1   

DEBUG: 16:33:23 - NewAzureRmAks begin processing with ParameterSet 'defaultParameterSet'.
DEBUG: 16:33:25 - using account id 'LukeBrennan@outlook.com'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'LukeBrennan@outlook.com', environment: 'AzureCloud', tenant:
 '1f89577f-4235-4965-83e5-aad1a7e761fc'
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://management.azure.com/subscriptions/b8e3a02e-456c-4060-a4fd-4fc9641f301d/resourceGroups/LukeBrennan-AZ-things/provid
ers/Microsoft.ContainerService/managedClusters/lukebrennan-akscluster?api-version=2017-08-31

Headers:
x-ms-client-request-id        : 64789c47-bc24-4960-b5f4-305385c00b21
accept-language               : en-US

Body:

DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
NotFound

Headers:
Pragma                        : no-cache
x-ms-failure-cause            : gateway
x-ms-request-id               : 22560fa6-dbbe-4223-8462-53232abc47c2
x-ms-correlation-request-id   : 22560fa6-dbbe-4223-8462-53232abc47c2
x-ms-routing-request-id       : AUSTRALIAEAST:20190213T053324Z:22560fa6-dbbe-4223-8462-53232abc47c2
Strict-Transport-Security     : max-age=31536000; includeSubDomains
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Wed, 13 Feb 2019 05:33:24 GMT

Body:
{
  "error": {
    "code": "ResourceNotFound",
    "message": "The Resource 'Microsoft.ContainerService/managedClusters/lukebrennan-akscluster' under resource group 'Luke
Brennan-AZ-things' was not found."
  }
}

VERBOSE: Cluster does not exist.
VERBOSE: Cluster is new.
VERBOSE: Preparing for deployment of your managed Kubernetes cluster.
VERBOSE: Using DNS name prefix LukeBrennan.
VERBOSE: Fetching SSH public key from file C:\Users\lukeb.SOUTHPACIFIC\.ssh\id_rsa.pub
VERBOSE: Deploying your managed Kubernetes cluster.
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/b8e3a02e-456c-4060-a4fd-4fc9641f301d/resourceGroups/LukeBrennan-AZ-things/provid
ers/Microsoft.ContainerService/managedClusters/lukebrennan-akscluster?api-version=2017-08-31

Headers:
x-ms-client-request-id        : fafe197b-5046-436f-a94d-1cc44e372b73
accept-language               : en-US

Body:
{
  "properties": {
    "dnsPrefix": "LukeBrennan",
    "kubernetesVersion": "1.8.1",
    "agentPoolProfiles": [
      {
        "name": "default",
        "count": 3,
        "vmSize": "Standard_D2_v2",
        "osDiskSizeGB": 0,
        "dnsPrefix": "LukeBrennan"
      }
    ],
    "linuxProfile": {
      "adminUsername": "azureuser",
      "ssh": {
        "publicKeys": [
          {
            "keyData": "ssh-rsa ##############################"
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "6e7416cd-cd1a-4995-8a7f-9eef47a8e28c",
      "secret": "#####################"
    }
  },
  "location": "australiaeast"
}

DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Pragma                        : no-cache
x-ms-ratelimit-remaining-subscription-writes: 1199
x-ms-correlation-request-id   : e557cadc-418d-4779-a999-bc04ff657558
x-ms-request-id               : 15df1b8a-0110-437a-8b30-430e1167c13a
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-routing-request-id       : AUSTRALIAEAST:20190213T053328Z:e557cadc-418d-4779-a999-bc04ff657558
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Wed, 13 Feb 2019 05:33:27 GMT
Server                        : nginx

Body:
{
  "code": "InvalidParameter",
  "message": "The value of parameter orchestratorProfile.OrchestratorVersion is invalid.",
  "target": "orchestratorProfile.OrchestratorVersion"
}

New-Azaks : The value of parameter orchestratorProfile.OrchestratorVersion is invalid.
At line:1 char:3
+   New-Azaks -ResourceGroupName $RG -name $AKSclustername -Location $S ...
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzAks], PSInvalidOperationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Aks.NewAzureRmAks
 
DEBUG: AzureQoSEvent: CommandName - New-AzAks; IsSuccess - False; Duration - 00:00:07.0578836; Exception - System.Managemen
t.Automation.PSInvalidOperationException: The value of parameter orchestratorProfile.OrchestratorVersion is invalid. ---> M
icrosoft.Rest.Azure.CloudException: The value of parameter orchestratorProfile.OrchestratorVersion is invalid.
   at Microsoft.Azure.Commands.Aks.Generated.Version2017_08_31.ManagedClustersOperations.<BeginCreateOrUpdateWithHttpMessag
esAsync>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Commands.Aks.Generated.Version2017_08_31.ManagedClustersOperations.<CreateOrUpdateWithHttpMessagesAsy
nc>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Commands.Aks.Generated.Version2017_08_31.ManagedClustersOperationsExtensions.<CreateOrUpdateAsync>d__
11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Commands.Aks.Generated.Version2017_08_31.ManagedClustersOperationsExtensions.CreateOrUpdate(IManagedC
lustersOperations operations, String resourceGroupName, String resourceName, ManagedCluster parameters)
   at Microsoft.Azure.Commands.Aks.NewAzureRmAks.<ExecuteCmdlet>b__4_0()
   at Microsoft.Azure.Commands.Aks.KubeCmdletBase.RunCmdLet(Action action)
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Commands.Aks.KubeCmdletBase.RunCmdLet(Action action)
   at Microsoft.Azure.Commands.Aks.NewAzureRmAks.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord();
DEBUG: Finish sending metric.
DEBUG: 16:33:33 - NewAzureRmAks end processing.
DEBUG: 16:33:33 - NewAzureRmAks end processing.

**Comment by maddieclayton** [comment]
@mboersma Can you take a look at this issue?

**Comment by hbagarich** [comment]
If I supply the Kubernetes version (1.12.4) to the command, everything works.
I guess the default version 1.8.1 is no longer supported by the Azure?

**Comment by mboersma** [comment]
@hbagarich yes, that's the problem. The error message `The value of parameter orchestratorProfile.OrchestratorVersion is invalid.` is correct but a bit hard to parse. AKS dropped support for provisioning new Kubernetes 1.8.x clusters recently.

You can see the currently supported versions using the CLI:

```
$ az aks get-versions -l eastus -o table
KubernetesVersion    Upgrades
-------------------  -----------------------
1.12.5               None available
1.12.4               1.12.5
1.11.7               1.12.4, 1.12.5
1.11.6               1.11.7, 1.12.4, 1.12.5
1.10.12              1.11.6, 1.11.7
1.10.9               1.10.12, 1.11.6, 1.11.7
1.9.11               1.10.9, 1.10.12
1.9.10               1.9.11, 1.10.9, 1.10.12
```

**Comment by lukeb1961** [comment]
still broken, five months later...

**Comment by cormacpayne** [comment]
@lukeb1961 we will update the default Kubernetes version used in our AKS cmdlets in the upcoming sprint. A temporary workaround can be found in [this comment](https://github.com/Azure/azure-powershell/issues/9573#issuecomment-509823626). I'm going to close this issue in favor of the one linked previously and we will track the status of the fix to the default version there.
